### PR TITLE
Implement add team member in user management page

### PR DESCRIPTION
**Pull request by jadmsaadaot**
*Issue #:*
https://github.com/bcgov/met-public/issues/1074

*Description of changes:*

- Add the add team member form
- Implement search as you type for users and engagements drop down
- search as you type in engagement form and user management pages

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

**Review by djnunez-aot — APPROVED** [review]

File: met-api/src/met_api/models/user.py [hunk 1]
```diff
@@ -34,21 +36,29 @@ class User(db.Model):  # pylint: disable=too-few-public-methods
     access_type = Column('access_type', String(200), nullable=True)
     status_id = db.Column(db.Integer, ForeignKey('user_status.id'))
 
+    @hybrid_property
```
Comment (tiagograf-AOT) [inline met-api/src/met_api/models/user.py:39]: neat

File: met-web/src/components/engagement/form/EngagementFormTabs/AddTeamMemberModal.tsx [hunk 2]
```diff
@@ -58,19 +51,19 @@ export const AddTeamMemberModal = () => {
...
         try {
             setUsersLoading(true);
-            const response = await getUserList();
+            const response = await getUserList({
+                search_text: searchText,
+            });
             setUsers(response.items);
             setUsersLoading(false);
         } catch (error) {
+            console.log(error);
```
Comment (tiagograf-AOT) [inline met-web/src/components/engagement/form/EngagementFormTabs/AddTeamMemberModal.tsx:66]: drop the console log

File: met-web/src/components/engagement/form/EngagementFormTabs/AddTeamMemberModal.tsx [hunk 3]
```diff
@@ -81,6 +74,12 @@ export const AddTeamMemberModal = () => {
         }
     };
 
+    const debounceLoadUsers = useRef(
+        debounce((searchText: string) => {
+            loadUsers(searchText);
```
Comment (tiagograf-AOT) [inline met-web/src/components/engagement/form/EngagementFormTabs/AddTeamMemberModal.tsx:79]: nice touch

File: met-web/src/components/userManagement/listing/AddUserModal.tsx [hunk 4]
```diff
@@ -48,25 +62,97 @@ export const AddUserModel = () => {
...
+        }
+        try {
+            setUsersLoading(true);
+            const response = await getUserList({
+                search_text: searchText,
+            });
+            setUsers(response.items);
+            setUsersLoading(false);
+        } catch (error) {
+            console.log(error);
```
Comment (tiagograf-AOT) [inline met-web/src/components/userManagement/listing/AddUserModal.tsx:89]: a couple more console logs here

**Review by tiagograf-AOT — APPROVED** [review]

File: met-web/src/components/userManagement/listing/AddUserModal.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jadmsaadaot) [reply met-web/src/components/userManagement/listing/AddUserModal.tsx:89]: removed

Comment (jadmsaadaot) [reply met-web/src/components/userManagement/listing/AddUserModal.tsx:89]: removed

File: met-web/src/components/engagement/form/EngagementFormTabs/AddTeamMemberModal.tsx (hunk 2 above)
Comment (jadmsaadaot) [reply met-web/src/components/engagement/form/EngagementFormTabs/AddTeamMemberModal.tsx:66]: removed

**Pull request merged**